jscomps/EmbedLiteSearchEngine.js

```diff
@@ -51,20 +51,17 @@ EmbedLiteSearchEngine.prototype = {
       }
       case "profile-after-change": {
         Services.obs.removeObserver(this, "profile-after-change");
-        Services.search.init(function addEngine_cb(rv) {
-            let engines = Services.search.getEngines({});
-            let engineNames = engines.map(function (element) {
-              return element.name;
-            });
-
-            let enginesAvailable = (engines && engines.length > 0);
-            var messg = {
-              msg: "init",
-              engines: engineNames,
-              defaultEngine: enginesAvailable && Services.search.defaultEngine ?
-                Services.search.defaultEngine.name : null
-            }
-            Services.obs.notifyObservers(null, "embed:search", JSON.stringify(messg));
+        Services.search.getEngines().then((engines) => {
```

Comment:
There's little function diff now, getEngines() call wait for init promise to complete.

You actually do this so that first you get init promise, upon init promise success, call getEngine

Reply:
I guess that we can go with this...